### DISCUSSION TITLE
.eslintrc with file extension when using npx next lint

### DISCUSSION THREAD

**Discussion by niko-achilles** [vercel/next.js#26857] [Ideas]
### Describe the feature you'd like to request

With `next.js` 11 the ability to configure `eslint` with a command `npx next lint` improves the developer experience for sure. 
`next.js command creates a .eslintrc file `without` file extension.  

Given `eslint`  marked as deprecated `eslintrc` file without file extension
and that i manually change the .eslintrc file extension to `.eslintrc.json` created by `npx next lint`
then it should be better if  `npx next lint` creates a .eslintrc.json file with the file extension `json`

### Describe the solution you'd like

command `npx next lint` --output--> .eslintrc.json file

### Describe alternatives you've considered

command `npx next lint` --output--> one of the .eslintrc.* file extensions supported by eslint as described here
https://eslint.org/docs/user-guide/configuring/configuration-files#configuration-file-formats

**Comment by housseindjirdeh** [comment]
Thanks for this! I haven't seen any official docs that suggest `.eslintrc` without a file extension is being deprecated (please share if you can), but you're right in that it isn't mentioned anywhere as a supported file format within their [documentation](https://eslint.org/docs/user-guide/configuring/configuration-files#configuration-file-formats). Outputting to 
a specific file format definitely sounds like the safest approach to use :)

Fix is here: #26884

**Reply by niko-achilles** [reply]
Sure , here: https://github.com/eslint/eslint/pull/4406#

if you look at the conversation, eslint fixed the `yaml` extension back on 2015 , and wish to support in a declarative way the file extensions. That`s the reason the supported file extensions are explicit documented in the documentation pages of eslint.
